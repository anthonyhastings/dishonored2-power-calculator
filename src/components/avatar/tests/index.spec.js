import renderer from 'react-test-renderer';
import Avatar from '../';

jest.mock(
  'Constants/character-slug-to-portrait',
  () => new Map([['mock-slug', 'mock-src']])
);

describe('Avatar component', () => {
  let testContext;

  const renderComponent = (props = {}) => (
    <Avatar {...testContext.defaultProps} {...props} />
  );

  beforeEach(() => {
    testContext = {};

    testContext.defaultProps = {
      name: 'Fake Name',
      slug: 'mock-slug',
    };
  });

  describe('when rendered', () => {
    beforeEach(() => {
      testContext.component = renderer.create(renderComponent());
    });

    it('renders a wrapping element containing an image', () => {
      expect(testContext.component).toMatchSnapshot();
    });
  });

  describe('when className is provided', () => {
    it('renders with extra className', () => {
      expect(renderer.create(renderComponent())).toMatchDiffSnapshot(
        renderer.create(
          renderComponent({
            className: 'mock-class',
          })
        )
      );
    });
  });
});
