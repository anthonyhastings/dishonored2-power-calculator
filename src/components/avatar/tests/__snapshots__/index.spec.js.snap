// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component when className is provided renders with extra className 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,7 +1,7 @@
  <div
-   className=\\"avatar\\"
+   className=\\"avatar mock-class\\"
  >
    <img
      alt=\\"Portrait of Fake Name\\"
      className=\\"avatar__portrait\\"
      draggable=\\"false\\""
`;

exports[`Avatar component when rendered renders a wrapping element containing an image 1`] = `
<div
  className="avatar"
>
  <img
    alt="Portrait of Fake Name"
    className="avatar__portrait"
    draggable="false"
    src="mock-src"
  />
</div>
`;
